### PR TITLE
Adds block production times page

### LINKED ISSUE #993: Expected vs actual block time
Going through @raedah's notes for material to include in the dev docs, saw this snipped on [Block production times](https://github.com/raedahgroup/dcrdevdocs/blob/master/start.md#block-production-times). This is taken from a post @davecgh made to Reddit, in response to being asked a few times regarding this. Worth including in dcrdocs?

### PR DESCRIPTION
**Pull request by s-ben**
This PR adds a new page on expected block times, using material from @davecgh's Reddit post ( Closes #993 ). 

@davecgh I've added a chart that should match the CDF you linked to, but needs technical review. Also, there were a couple places I wanted to spell out details more clearly, but lacked some numbers. Not necessary, but nice to haves.

> However, in Decred, this calculation is also interpolated into Bitcoin-like difficulty window periods. 

How exactly does this work? Do we just average only those blocks falling within a window? What is the size of the window? X number of blocks? What is the power of the exponential weighting? Do we have a formula we can provide?

>Note that the 0 to 10 and 0 to 30 second intervals slightly underperform the ideal values, which is  expected because miners must wait for the votes to arrive before they can start building a new block, and this fact slightly skews the number of blocks found within the time spans downwards.

What votes exactly are miners waiting on before they can start building a new block? The PoS validators validating the last PoW block?

### CODE REVIEW

File: docs/research/block-production-times.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# <img class="dcr-icon" src="/img/dcr-icons/PoWMine.svg" /> Block Production Times 
+
+---
+
+Decred’s difficulty adjusts such that new blocks are found every five minutes on average. The Proof-of-Work (PoW) difficulty is calculated from the exponentially weighted average of differences in previous block times, similar to PeerCoin[^1]. However, in Decred, this calculation is also interpolated into Bitcoin-like difficulty window periods. 
+
+The time between blocks can be modeled as a Poisson process ($\Lambda$ =1, k =0) . Because this is a special case of a [once in an interval](https://en.wikipedia.org/wiki/Poisson_distribution#Once_in_an_interval_events:_The_special_case_of_%CE%BB_=_1_and_k_=_0) event, the probability of a block being produced in a given time after the last block was produced can be modeled with a [Cumulative Distribution Function (CDF)](https://en.wikipedia.org/wiki/Cumulative_distribution_function) for an [exponential distribution](https://en.wikipedia.org/wiki/Exponential_distribution).
```
Comment (jholdstock) [inline docs/research/block-production-times.md]: `($\Lambda=1$, $k=0$)`

File: docs/research/block-production-times.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+| 5 min   | 63.21%   | 63.74% |
+| 6 min   | 69.88%   | 70.38% |
+| 7 min   | 75.34%   | 75.74% |
+| 10 min  | 86.47%   | 86.72% |
+| 20 min  | 98.17%   | 98.09% |
+| 30 min  | 99.75%   | 99.70% |
+
+The figure below shows shows the actual values plotted against the CDF used to derive expected percentages.
+
+![block times chart](../img/block_times_chart.png)
```
Comment (jholdstock) [inline docs/research/block-production-times.md]: Image paths should be absolute:

`block times chart`

File: docs/research/block-production-times.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+As we can see, it turns out that the actual distribution is quite close to the expected perfect distribution, which shows the network is operating in a healthy fashion. Note that the 0 to 10 and 0 to 30 second intervals slightly underperform the ideal values, which is  expected because miners must wait for the votes to arrive before they can start building a new block, and this fact slightly skews the number of blocks found within the time spans downwards.
+
+
+## :fa-book: References
+
+[^1]: King S. and Nadal S. 2012. [PPCoin: Peer-to-peer crypto-currency with proof-of-stake](https://decred.org/research/king2012.pdf).
```
Comment (jholdstock) [inline docs/research/block-production-times.md]: Fix caps here as per the actual papers title: `PPCoin: Peer-to-Peer Crypto-Currency with Proof-of-Stake`

**Comment by s-ben** [comment]
@jholdstock comments addressed. Unless @davecgh or others have changes should be good to go.

File: docs/research/block-production-times.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+| 7 min   | 75.34%   | 75.74% |
+| 10 min  | 86.47%   | 86.72% |
+| 20 min  | 98.17%   | 98.09% |
+| 30 min  | 99.75%   | 99.70% |
+
+The figure below shows shows the actual values plotted against the CDF used to derive expected percentages.
+
+![block times chart](/img/block_times_chart.png)
+
+As we can see, it turns out that the actual distribution is quite close to the expected perfect distribution, which shows the network is operating in a healthy fashion. Note that the 0 to 10 and 0 to 30 second intervals slightly underperform the ideal values, which is  expected because miners must wait for the votes to arrive before they can start building a new block, and this fact slightly skews the number of blocks found within the time spans downwards.
```
Comment (davecgh) [inline docs/research/block-production-times.md]: is <double space> expected

**Pull request merged**